veps/vep-3.md

```diff
@@ -0,0 +1,38 @@
+# VEP-2 Vitess Upgrade Order
+
+```
+VEP: 3
+Title: Vitess Upgrade Order
+Author: Sugu Sougoumarane
+Reviewers: Deepthi Sigireddi, Rafael Chacon Vivas
+Status: Proposed
+Created: 2020-07-14
+```
+
+## Abstract
+
+Vitess is predominantly a two layer system. At the top layer, we have vtgate, vtctld and vtworker. At the next layer, we have vttablet which connects to mysql.
+In Kubernetes, we also use mysqlctld as a sidecar to manage the life cycle of the underlying mysql. This would belong to the vttablet layer.
+
+Vitess also relies on an externally configured "topo server", which can be configured and upgraded independently.
+
+In terms of interaction, the top layer sends requests to the vttablets. The vttablets send messages to each other. VTTablets also send messages to mysqlctld.
```

Comment:
consistent naming: vttablets or VTTablets but not both. The second one might be better because we use it in website docs.

Reply:
VTTablet is the capitalized version that I've been using at the beginning of sentences. Won't it look odd if we used the uncapitalized version?

Reply:
Yeah I was suggesting we should use VTTablet everywhere. But it's fine to use all lowercase when it is not the beginning of the sentence, I guess.